test/helpers/kubectl.go

```diff
@@ -4073,3 +4073,22 @@ func logGathererSelector(allNodes bool) string {
 
 	return selector
 }
+
+// GetDNSProxyPort returns the port the Cilium DNS proxy is listening on
+func (kub *Kubectl) GetDNSProxyPort(ciliumPod string) int {
+	const pickDNSProxyPort = `iptables-save | grep "udp.*TPROXY to host cilium-dns-egress" | sed 's/^.*--on-port \([1-9][0-9]*\).*$/\1/'`
```

Comment:
```suggestion
	const pickDNSProxyPort = `iptables-save | sed -n '/udp.*TPROXY to host cilium-dns-egress/ s/.*--on-port \([1-9][0-9]*\).*/\1/p'`
```

You could get rid of the `grep`, and of the `^`/`$` anchors in the pattern. If you know the name of the iptables table where the rule is located (and if it's constant), we could maybe pass `-t <name>` to `iptables-save`. We probably won't save much, but heh.

Reply:
Nice, will do :-)